rotkehlchen/chain/evm/decoding/giveth/decoder.py

```diff
@@ -0,0 +1,142 @@
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.giv_token_id = giv_token_id
+        self.pow_token_id = pow_token_id
+        self.distro_address = distro_address
+        self.givpower_staking_address = givpower_staking_address
+
+    def _decode_token_locked(self, context: DecoderContext) -> DecodingOutput:
+        if not self.base.is_tracked(user := bytes_to_address(context.tx_log.topics[1])):
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[:32]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,  # GIV has 18 decimals
+        )
+        rounds = int.from_bytes(context.tx_log.data[32:64])
+
+        lock_event = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=Asset(self.giv_token_id),
+            balance=Balance(amount),
+            location_label=user,
+            notes=f'Lock {amount} GIV for {rounds} round/s',
+            address=context.tx_log.address,
+            counterparty=CPT_GIVETH,
+        )
+        receive_event = None
+        for event in context.decoded_events:
+            if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.asset.identifier == self.pow_token_id and
+                    event.location_label == user
+            ):
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.counterparty = CPT_GIVETH
+                event.notes = f'Receive {event.balance.amount} POW after locking GIV'
+                receive_event = event
+                break
+
+        else:
+            log.error(f'Could not find the GivPoW token transfer after locking GIV for {context.transaction}')  # noqa: E501
+
+        maybe_reshuffle_events(
+            ordered_events=[lock_event, receive_event],
+            events_list=context.decoded_events,
+        )
+        return DecodingOutput(event=lock_event)
+
+    def decode_claim(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] != SIMPLE_CLAIM:
+            return DEFAULT_DECODING_OUTPUT
+
+        if not self.base.is_tracked(user := bytes_to_address(context.tx_log.topics[1])):
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[:32]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,  # GIV has 18 decimals
+        )
+        for event in context.decoded_events:
+            if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.asset.identifier == self.giv_token_id and
```

Comment:
comparing the asset will also compare the identifier right? but not really an important thing

Reply:
we only have identifiers here. This save us unecessary asset instantiations.

Reply:
I meant that you can do 

```
event.asset == self.giv_token_id
```

due to the overload in the eq operator